tasks/main.yml

```diff
@@ -10,20 +10,29 @@
     use: "{{ (__snapshot_is_ostree | d(false)) |
               ternary('ansible.posix.rhel_rpm_ostree', omit) }}"
 
-- name: Show snapshot command
-  debug:
-    var: __snapshot_cmd
-    verbosity: 2
-
-- name: Run snapshot command and handle errors
+- name: Run snapshot module and handle errors
   block:
-    - name: Run snapshot command {{ snapshot_lvm_action }}
-      ansible.builtin.script: "{{ __snapshot_cmd }}"
-      args:
-        executable: "{{ __snapshot_python }}"
+    - name: Run snapshot module {{ snapshot_lvm_action }}
+      snapshot:
+        snapshot_lvm_action: "{{ snapshot_lvm_action }}"
+        ansible_check_mode: "{{ ansible_check_mode | d(false) }}"
+        snapshot_lvm_all_vgs: "{{ snapshot_lvm_all_vgs | d(false) }}"
+        snapshot_lvm_verify_only: "{{ snapshot_lvm_verify_only | d(false) }}"
+        snapshot_lvm_mount_origin: "{{ snapshot_lvm_mount_origin | d(false) }}"
+        snapshot_lvm_mountpoint_create: >-
+          "{{ snapshot_lvm_mountpoint_create | d(false) }}"
+        snapshot_lvm_unmount_all: "{{ snapshot_lvm_unmount_all | d(false) }}"
+        snapshot_lvm_percent_space_required:
+          "{{ snapshot_lvm_percent_space_required | d(omit) }}"
+        snapshot_lvm_vg: "{{ snapshot_lvm_vg | d(omit) }}"
+        snapshot_lvm_lv: "{{ snapshot_lvm_lv | d(omit) }}"
+        snapshot_lvm_snapset_name: "{{ snapshot_lvm_snapset_name | d(omit) }}"
+        snapshot_lvm_mount_options: "{{ snapshot_lvm_mount_options | d(omit) }}"
+        snapshot_lvm_fstype: "{{ snapshot_lvm_fstype | d(omit) }}"
+        snapshot_lvm_mountpoint: "{{ snapshot_lvm_mountpoint | d(omit) }}"
+        snapshot_lvm_set: "{{ snapshot_lvm_set | to_json }}"
```

Comment:
@richm is there a better way to pass snapshot_lvm_set?

I'd like it to be None if it is not set, but the way I've done it here I'm getting "{}\0" when I read it in the python.

Reply:
Then if you define `snapshot_lvm_set` as a `list` of `dict` in the module, you can just use
```yaml
        snapshot_lvm_set: "{{ snapshot_lvm_set | d(omit) }}"
```

Reply:
So, on the module side, I will remove the code that expects JSON and use dict()?

When I define it as you suggested, the module gets snapshot_lvm_set set to a string, right?  But the string is no longer JSON (the values have single quotes).

Reply:
> So, on the module side, I will remove the code that expects JSON and use dict()?

yes
 
> When I define it as you suggested, the module gets snapshot_lvm_set set to a string, right?

No, the module should get `snapshot_lvm_set` as a dict.

> But the string is no longer JSON (the values have single quotes).

in defaults/main.yml you should have
```yaml
snapshot_lvm_set: {}
```